src/game/Game.java

```diff
@@ -350,51 +356,35 @@ public void undoMove() {
 		this.numMoves = 4;
 	}
 
-	public boolean equals(Game compGame) {
-		// for(int i = 0; i < this.moves.size(); i++) {
-		// if(!this.moves.get(i).equals(compGame.moves.get(i))) {
-		// return false;
-		// }
-		// }
-		if (!this.moves.equals(compGame.moves)) {
+	@Override
+	public boolean equals(Object o) {
+		if (!(o instanceof Game)) {
 			return false;
 		}
+		Game g = (Game) o;
+		boolean historyEqual = this.moves.equals(g.getMoves()) && this.currentBoard.equals(g.getBoardState())
+				&& this.turnNumber == g.getTurnNumber() && this.turnCounter == g.getTurnCounter();
 
-		if (!this.currentBoard.equals(compGame.currentBoard)) {
-			return false;
-		}
+		boolean p1Equal = this.p1Name.equals(g.getP1Name());
+		boolean p2Equal = this.p2Name.equals(g.getP2Name());
 
-		if (this.turnNumber != compGame.turnNumber) {
-			return false;
-		}
-		if (this.moveTimer != compGame.moveTimer) {
-			return false;
-		}
-		if (this.p1TimeBank != compGame.p1TimeBank) {
-			return false;
-		}
-		if (this.p2TimeBank != compGame.p2TimeBank) {
-			return false;
-		}
-		if (this.turnCounter != compGame.turnCounter) {
-			return false;
-		}
-		if (!this.p1Name.equals(compGame.p1Name)) {
-			return false;
-		}
-		if (!this.p2Name.equals(compGame.p2Name)) {
-			return false;
-		}
-		if (this.winner != compGame.winner) {
-			return false;
-		}
-		if (this.numMoves != compGame.numMoves) {
-			return false;
-		}
-		if (this.playerTurn != compGame.playerTurn) {
-			return false;
-		}
+		boolean turnEqual = this.playerTurn == g.getPlayerTurn() && this.numMoves == g.getNumMoves()
+				&& this.winner == g.getWinner();
+
+		boolean timerEqual = this.moveTimer == g.getMoveTimer();
+		return timerEqual && turnEqual && p1Equal && p2Equal && historyEqual;
+	}
+
+	@Override
+	public int hashCode() {
+		return this.moves.hashCode() + this.currentBoard.hashCode() + getHashCode(this.turnNumber)
+				+ getHashCode(this.turnCounter) + this.p1Name.hashCode() + this.p2Name.hashCode()
+				+ getHashCode(this.playerTurn) + getHashCode(this.numMoves) + getHashCode(this.winner)
+				+ getHashCode(this.moveTimer);
+	}
 
-		return true;
+	// this prevents collisions better with serial numbers
+	private int getHashCode(int numb) {
+		return Integer.rotateLeft(numb, Integer.BYTES / 2);
```

Comment:
this is better at preventing collisions if we expect integers to be near each other, which we do